### DISCUSSION TITLE
JDBC 中的用户名、密码验证是如何处理的？

### DISCUSSION THREAD

**Discussion by fuhaiq** [apache/kyuubi#2853] [Q&A] [answered]
kyuubi 是如何验证 JDBC 连接中的 username、password的？
```
$ bin/beeline --help
Usage: java org.apache.hive.cli.beeline.BeeLine
   -u <database url>               the JDBC URL to connect to
   -r                              reconnect to last saved connect url (in conjunction with !save)
   -n <username>                   the username to connect as
   -p <password>                   the password to connect as
```
我看到 [安全文档上](https://kyuubi.apache.org/docs/latest/security/authentication.html) 有 kyuubi.authentication 配置：

- NOSASL: raw transport.
- NONE: no authentication check.
- KERBEROS: Kerberos/GSSAPI authentication.
- LDAP: Lightweight Directory Access Protocol authentication.

目前已经完成 kerberos 的认证，但我感觉这个不是 kyuubi 负责的吧？kyuubi 有没有像 mysql 那样的用户认证？认证成功后 JDBC 能建立

**Comment by wForget** [comment, marked as answer]
The [authentication configuration document](https://kyuubi.apache.org/docs/latest/deployment/settings.html#authentication) is more detailed, you can set `kyuubi.authentication=CUSTOM` and configure a user-defined implementation of `PasswdAuthenticationProvider` for `kyuubi.authentication.custom.class`.

**Comment by pan3793** [comment]
A typical Kerberos authentication does not use username/password, but TGT, that's why you should run `kinit` before using beeline to connect Kyuubi.

To use username/password for authentication, LDAP is an out-of-box solution. CUSTOM is also an option.

**Reply by fuhaiq** [reply]
谢谢～
